lib/src/main/java/graphql/nadel/engine/transform/NadelTypeRenameResultTransform.kt

```diff
@@ -77,13 +77,16 @@ internal class NadelTypeRenameResultTransform : NadelTransform<State> {
                 underlyingTypeName = underlyingTypeName,
             )
 
-            val typeName: String = if (overallField.objectTypeNames.contains(overallTypeName)) {
+            val typeName: String = if (
+                !executionContext.hints.sharedTypeRenames(service) ||
+                overallField.objectTypeNames.contains(overallTypeName)
+            ) {
                 overallTypeName
             } else {
-                overallField.objectTypeNames.single {
```

Comment:
that `single` blew up here during the rollout, handling it safer by defaulting to `overallTypeName`